components/CreateClassModal/CreateClassModal.js

```diff
@@ -0,0 +1,169 @@
+import LocalizationProvider from '@mui/lab/LocalizationProvider';
+import MainButton from '../basecomponents/MainButton.js'
+import styles from '../../utils/styles/CreateClassModalStyles/CreateClassModalStyles.module.css';
+
+
+export default function CreateClassModal() {
+  const [subject, setSubject] = useState('');
+  const [type, setType] = useState('');
+  const [newStartTime, setNewStartTime] = useState(null);
+  const [newEndTime, setNewEndTime] = useState(null);
+  const [classObj, setClassObj] = useState({});
+  const [open, setOpen] = useState(false);
+
+  const handleChange = (e) => {
+    if(e.target.name === 'capacity') {
+      setClassObj({ ...classObj, [e.target.name]: Number(e.target.value) })
+    } else {
+      setClassObj({ ...classObj, [e.target.name]: e.target.value })
+    }
+  }
+
+  return (
+    <div>
+      <MainButton value="Create a Class" onClick={() => setOpen(true)} />
+      <Dialog open={open} onClose={() => setOpen(false)}>
+        <DialogTitle className={styles.formHeader}>Create a Class</DialogTitle>
+        <DialogContent>
+          {/* ----------------------COURSE NAME-------------------------- */}
+          <FormControl component="fieldset" required className={styles.classForm}>
+            <TextField
+              required
+              name="name"
+              label="New Course Name"
+              type="text"
+              fullWidth
+              color='primary'
+              onChange={handleChange}
+              placeholder="E.g. Calculus II"
+            />
+            {/* ----------------------COURSE DESCRIPTION-------------------------- */}
+            <TextField
+              required
+              className={styles.classForm}
+              multiline
+              rows={4}
+              name="description"
+              margin="dense"
+              label="Course Description"
+              type="text"
+              color='primary'
+
+              onChange={handleChange}
+              placeholder="E.g. Advanced Level Calculus for those looking to build upon their Calculus fundamentals."
+            />
+            {/* ----------------------SELECT SUBJECT-------------------------- */}
+            <FormControl className={styles.subjectSelector}>
+              <InputLabel>Select a Subject</InputLabel>
+              <Select
+                required
+                label="Select a Subject"
+                defaultValue="Select a subject"
+                value={subject}
+                name="subject"
+                onChange={(e) => {
+                  handleChange(e);
+                  setSubject(e.target.value);
+                }}
+              >
+                <MenuItem value="Math">Math</MenuItem>
+                <MenuItem value="Science">Science</MenuItem>
+                <MenuItem value="History">History</MenuItem>
+                <MenuItem value="Literature">Literature</MenuItem>
+                <MenuItem value="Language">Language</MenuItem>
```

Comment:
Could keep an object like "subjectsOffered" in constants and map over that here. That way in future iterations where our app may offer additional subjects we wouldn't need to hard code an extra subject here, could just add in one spot. We would also need to refactor some other components to map over that "subjectOffered" object in past made components.